sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/ResumeFromBackgroundTask.java

```diff
@@ -16,14 +23,35 @@
     @SuppressLint("StaticFieldLeak")
     private final Context mContext;
 
-    ResumeFromBackgroundTask(@NonNull Context context) {
+    private final long mDownloadedId;
+
+    ResumeFromBackgroundTask(@NonNull Context context, long downloadedId) {
         mContext = context;
+        mDownloadedId = downloadedId;
     }
 
     @Override
-    protected Void doInBackground(Void... voids) {
+    protected Void doInBackground(Void[] args) {
         Distribute distribute = Distribute.getInstance();
+
+        /*
+         * Completion might be triggered in background before AppCenter.start
+         * if application was killed after starting download.
+         *
+         * We still want to generate the notification: if we can find the data in preferences
+         * that means they were not deleted, and thus that the sdk was not disabled.
+         */
         distribute.startFromBackground(mContext);
+
+        /* Check download id is what we expected. */
+        AppCenterLog.debug(LOG_TAG, "Check download id=" + mDownloadedId);
+        long expectedDownloadId = SharedPreferencesManager.getLong(PREFERENCE_KEY_DOWNLOAD_ID, INVALID_DOWNLOAD_IDENTIFIER);
+        if (expectedDownloadId == INVALID_DOWNLOAD_IDENTIFIER || expectedDownloadId != mDownloadedId) {
```

Comment:
Develop also has the bug but just FYI, this is not using the distribute lock so this storage state could change in another thread. No need to fix it in this PR though as develop has that too.

Reply:
Yeah, let's get this one merged. I can do the fix separately

Reply:
We may file a bug, it's not part of the feature anyway.